gordon_janitor_gcp/http_client.py

```diff
@@ -62,119 +58,31 @@
-        if not scopes:
-            scopes = ['cloud-platform']
-        return [self.SCOPE_TMPL_URL.format(scope=s) for s in scopes]
-
-    def _load_credentials(self):
-        # TODO (lynn): FEATURE - load other credentials like app default
-        return service_account.Credentials.from_service_account_info(
-            self._keydata, scopes=self.scopes)
-
-    def _setup_token_request(self):
-        url = self._keydata.get('token_uri')
-        headers = {
-            'Content-type': 'application/x-www-form-urlencoded',
-        }
-        body = {
-            'assertion': self._creds._make_authorization_grant_assertion(),
-            'grant_type': self.JWT_GRANT_TYPE,
-        }
-        body = urllib.parse.urlencode(body)
-        return url, headers, bytes(body.encode('utf-8'))
-
-    async def refresh_token(self):
-        """Refresh oauth access token attached to this HTTP session.
-
-        Raises:
-            exceptions.GCPAuthError: if no token was found in the
-                response.
-            exceptions.GCPHTTPError: if any exception occurred.
-        """
-        url, headers, body = self._setup_token_request()
-
-        logging.debug(REQ_LOG_FMT.format(method='POST', url=url))
-        async with self._session.post(url, headers=headers, data=body) as resp:
-            log_kw = {
-                'method': 'POST',
-                'url': url,
-                'status': resp.status,
-                'reason': resp.reason,
-            }
-            logging.debug(RESP_LOG_FMT.format(**log_kw))
-
-            # avoid leaky abstractions and wrap http errors with our own
-            try:
-                resp.raise_for_status()
-            except aiohttp.ClientResponseError as e:
-                msg = f'Issue connecting to {resp.url.host}: {e}'
-                logging.error(msg, exc_info=e)
-                raise exceptions.GCPHTTPError(msg)
-
-            response = await resp.json()
-            try:
-                self.token = response['access_token']
-            except KeyError:
-                msg = 'No access token in response.'
-                logging.error(msg)
-                raise exceptions.GCPAuthError(msg)
 
-        self.expiry = _client._parse_expiry(response)
+    def __init__(self, auth_client=None, session=None):
+        self._creds = auth_client.creds
+        self._session = session or auth_client._session
 
     async def set_valid_token(self):
         """Check for validity of token, and refresh if none or expired."""
         is_valid = False
 
-        if self.token:
+        if self._creds.token:
             # Account for a token near expiration
             now = datetime.datetime.utcnow()
             skew = datetime.timedelta(seconds=60)
-            if self.expiry > (now + skew):
+            if self._creds.expiry > (now + skew):
                 is_valid = True
 
         if not is_valid:
-            await self.refresh_token()
+            await self._creds.refresh_token()
```

Comment:
I was going through where `creds` are used and in this instance I think this was supposed to be `await auth_client.refresh_token()` since google's Credential object doesn't have an async `refresh_token` method.